securityconfig/roles.yml

```diff
@@ -61,3 +61,35 @@ anomaly_full_access:
         - 'indices_monitor'
         - 'indices:admin/aliases/get'
         - 'indices:admin/mappings/get'
+
+# Allows users to read and download Reports
+reports_instances_read_access:
+  reserved: true
+  cluster_permissions:
+    - 'cluster:admin/opendistro/reports/instance/get_all'
+    - 'cluster:admin/opendistro/reports/instance/get'
```

Comment:
What the difference between`get` and `get_all` ? Why we have to two API?

Reply:
yes they are 2 different APIs. thinking again may be it makes sense to rename get_all to list

Reply:
Standard way is have action as `cluster:admin/opendistro/reports/instance/get` and API like `GET reports\{id}` gets single report `GET \reports` get all reports. You also can support filters as required.  This is how we do on Alerting: https://github.com/opendistro-for-elasticsearch/alerting/blob/master/alerting/src/main/kotlin/com/amazon/opendistroforelasticsearch/alerting/resthandler/RestGetDestinationsAction.kt

Reply:
How to support response uniformity with pagination support? Initially we had the same way but then changed to 2 separate APIs.

Reply:
example: https://restfulapi.net/http-methods/#get